Documentation/cockroachdb-cluster-crd.md

```diff
@@ -0,0 +1,61 @@
+---
+title: CockroachDB Cluster
+weight: 37
+indent: true
+---
+
+# CockroachDB Cluster CRD
+
+CockroachDB database clusters can be created and configuring using the `clusters.cockroachdb.rook.io` custom resource definition (CRD).
+Please refer to the the [user guide walk-through](cockroachdb.md) for complete instructions.
+This page will explain all the available configuration options on the CockroachDB CRD.
+
+## Sample
+
+```yaml
+apiVersion: cockroachdb.rook.io/v1alpha1
+kind: Cluster
```

Comment:
remember that the full name of the type includes the group as well, so the full name is `cluster.cockroachdb.rook.io`.  Creating an instance of this type won't create just a database, it will create a database cluster.  The cockroach docs refer to it as a cluster also:
> This page shows you how to orchestrate the deployment and management of a secure 3-node CockroachDB cluster

If you guys are strongly in favor of not naming the type cluster, I'm open to it.  Although I don't think any further qualifiers are needed, such as `sqlDatabase`, because there is only 1 type of cockroachdb database as far as I know. I don't think the sql prefix would be necessary.  Thoughts?

Reply:
@jbw976 since cockroachdb docs refer to it as a cluster, i'm ok with it as `cluster` instead of 'database`. However, it still concerns me that we will have multiple `cluster` types even if we can fully qualify them in the docs. If we always require them to use the fully qualified name, they would have to type:
```
kubectl get clusters.ceph.rook.io
```
Yet they would also be tempted to still use the command
```
kubectl get clusters
```
Which would fail if other clusters happened to be defined. In the case of rook containing multiple `cluster` types, it becomes much more likely to hit this issue.

Take the etcd operator as another example. It defines crd types that have `etcd` in the `kind` name even though `etcd` is also in the group name. [This example](https://github.com/coreos/etcd-operator/blob/master/example/etcd-restore-operator/restore_cr.yaml#L2) shows a kind of `EtcdRestore` inside the group `etcd.database.coreos.com`. While it is certainly redundant to use the fully qualified name, the common case is to use only the kind name. This makes it clear when you get it that it is an etcd type and there will never be a conflict to worry about.
```
$ kubectl get etcdcluster
```

This makes me think we should define non-ambiguous kind names such as `CephCluster` and `CockroachDBCluster`.